### PR TITLE
Add tests for ANSI SGR state that crosses line boundaries

### PR DESCRIPTION
**Pull request by tempoz**
Tests for #10016

### CODE REVIEW

File: app/terminal/text_test.ts [hunk 1]
```diff
@@ -23,3 +23,79 @@ describe("normalizeSpace", () => {
     expect(normalizeSpace("\x1b[33m\t12345\x1b[0m\t12345")).toEqual("\x1b[33m        12345\x1b[0m   12345");
   });
 });
+
+describe("getContent", () => {
+  it("should handle ANSI SGR state that persists across lines", () => {
+    expect(getContent("\x1b[32mMulti-line\nColor\x1b[m\nReset", "", 0).rows).toEqual([
```
Comment (bduffany) [inline app/terminal/text_test.ts:29]: nit: maybe also copy this test but without the `\n` before `Reset`, to test the offset/length logic within a single line

Comment (tempoz) [reply app/terminal/text_test.ts:29]: Done!
